c/linkedlist.h

```diff
@@ -0,0 +1,189 @@
+// Define linked list data structure
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+// Variables for random node generation
+#define NODE_STRING 10
+#define NODE_NUM 100
+
+// Defines node structure that list will hold
+typedef struct node {
+    char *string; 
+    int num;
+    struct node *next;
+} node; 
+
+/* ------ FUNCTIONS ------ */ 
+// Display contents of list
+// field = "string", "num", or "all"
+void printList(node **head, char *field) {
+    node *iterator = *head;
+    if (strcmp(field, "num") == 0) {
+        while (iterator != NULL) {
+            printf("%i -> ", iterator->num);
+            iterator = iterator->next;
+        }
+        printf("End\n");
+        return;
+    } else if (strcmp(field, "string") == 0) { 
```

Comment:
I wonder what other patterns you could employ to accomplish printing no matter what kind of data structure backs your node.... :)